### DISCUSSION TITLE
`code-server` command

### DISCUSSION THREAD

**Discussion by aghasemi** [coder/code-server#4959] [Q&A]
Hi,

I use `code-server` as a shell command inside the IDE in some cases, e.g. to open files. It opens the file in a new tab in the IDE. Also when calling it on a folder, for example, it opens the folder in a new Browser tab.

What I realised is that, if I have multiple open browser tabs running CS, the file is opened in all of them. And even more interesting, if I have CS open in two browsers on two different devices, running `code-server ` with a folder as the argument, will open a new browser tab _on both devices_! 

Am I right and is this the expected behaviour? Are there documents/practices on using code-server as a shell command?

**Comment by jsjoeio** [comment]
Very interesting 🤔 

> It opens the file in a new tab in the IDE. 

This is expected. 

> Also when calling it on a folder, for example, it opens the folder in a new Browser tab.

This is also expected AFAIK.

> What I realised is that, if I have multiple open browser tabs running CS, the file is opened in all of them. And even more interesting, if I have CS open in two browsers on two different devices, running code-server with a folder as the argument, will open a new browser tab on both devices!

This is not something I would have expected 🤔 It's kinda cool but not expected 😂 Thoughts @code-asher? We may need to open a  bug report.

**Reply by code-asher** [reply]
Interesting!  I would definitely expect it to only open in the current browser tab, so I doubt this was intended.  I think upstream has their own implementation of this so it would be worth looking into using theirs to see if that resolves the issue, otherwise we might need to dig deeper.
